### DISCUSSION TITLE
[Enhancement] [select-v2] Select V2组件的下拉弹出层宽度问题

### DISCUSSION THREAD

**Discussion by Ycq428071** [element-plus/element-plus#14466] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `select-v2`

## Description

当下拉框选项的label值较长时，由于Select V2组件的弹出层width是固定的，label值溢出的文字省略号显示了，无法看到完整的label值；尝试直接在谷歌浏览器devtools中修改弹出层width值后，发现鼠标移动切换hover选项时，width会变回原来的值，得出结论无法通过修改css解决该问题；然后观察了一下Select组件发现两者的在弹出层宽度的css属性使用上有所区别，Select组件中用的是min-width（弹出层宽度可以直接被内容撑开，不会有文本溢出省略的情况），而Select V2组件用的是width宽度被固定死了；建议是在Select V2组件中也使用min-width让弹出层宽度被内容撑开，或者当出现省略号时，能有一个tooltip用来显示完整的label值（个人认为还是前者更好）。

### Additional comments

顺便提一下，也在Select V2组件试过popper-options这个属性，应该是个bug，设置了placement: 'top-start'后，弹出层会在输入框上方出现一瞬间，然后立马回到下面

![image](https://github.com/element-plus/element-plus/assets/115223274/5aecd97a-753f-41c8-afa0-26072d16e89a)

**Comment by kooriookami** [comment]
我考虑过这个问题，select-v2是虚拟下拉，没有办法计算出所有Option最大的一个宽度。

**Reply by shizhihuaxu** [reply]
可否提供一个选项用于设定一个固定宽度呢

**Comment by shizhihuaxu** [comment]
@Ycq428071 请问最后怎么解决的设定宽度的问题

**Reply by Ycq428071** [reply]
可以二次封装下 tooltip 组件，然后用自定义 option 的插槽，组件里面判断下如果文本溢出就提示完整内容；
我当时还是用回 select 组件了，封装了下模糊查询和滚动加载分页数据的逻辑，复用的基础列表查询接口。
